services/src/main/scala/cromwell/services/womtool/impl/WomtoolServiceInCromwellActor.scala

```diff
@@ -0,0 +1,94 @@
+package cromwell.services.womtool.impl
+
+import akka.actor.{Actor, ActorRef, Props}
+import akka.pattern.pipe
+import cats.data.Validated.{Invalid, Valid}
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import cromwell.core.Dispatcher.ServiceDispatcher
+import cromwell.core.WorkflowSourceFilesCollection
+import cromwell.languages.util.ImportResolver.HttpResolver
+import cromwell.languages.LanguageFactory
+import cromwell.languages.util.{ImportResolver, LanguageFactoryUtil}
+import cromwell.services.womtool.WomtoolServiceMessages._
+import cromwell.util.GracefulShutdownHelper.ShutdownCommand
+import wom.core.WorkflowSource
+import wom.expression.NoIoFunctionSet
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class WomtoolServiceInCromwellActor(serviceConfig: Config, globalConfig: Config, serviceRegistryActor: ActorRef) extends Actor with LazyLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  override def receive: Receive = {
+    case DescribeRequest(filesCollection) =>
+
+      // We are consciously wrapping a Future around the Await.result way down in the HTTP import resolver until we can update the whole call hierarchy to async
+      // https://doc.akka.io/docs/akka/2.5.16/actors.html?language=scala#ask-send-and-receive-future
+      Future {
+        describeWorkflow(filesCollection)
+      } pipeTo sender()
```

Comment:
I know you've discussed it already. But just as a "ToL possibly going wrong scenario", since you run this actor on the service dispatcher (https://github.com/broadinstitute/cromwell/pull/4467/files#diff-b58faeefe405210df3e6d6c858cf66a9R93) and this schedules the `describeWorkflow` work on that dispatcher and immediately proceeds to process the next message in the mailbox, you could potentially flood the dispatcher (and its underlying execution context) with `describeWorkflow` work, which would impact all other services, including metadata and potentially slow everything down especially if `describeWorkflow(filesCollection)` is somewhat resource consuming or blocking for a non trivial amount of time.
If you've thought about this already then ignore this comment :)

Reply:
You're absolutely right about the potential dangers, I've added an extra warning to Swagger (in addition to changelog)
>This endpoint is undergoing active development. Its performance has not been validated and it **should not be called on a production instance of Cromwell**.